state/merkle.go

```diff
@@ -61,16 +53,18 @@ func (s *State) Commit(version uint64) ([]byte, error) {
 		return nil, err
 	}
 
+	// store a new version
 	var hash []byte
-	if s.persistent {
-		if s.committed.Tree.Size() > 0 || s.committed.Tree.LatestVersion() > 0 {
-			hash, err = s.committed.Tree.SaveVersion(version)
-			if err != nil {
-				return nil, err
-			}
+	if s.committed.Tree.Size() > 0 || s.committed.Tree.LatestVersion() > 0 {
```

Comment:
I wrote this, but I don't like it that much - it doesn't communicate the intent very well. I wonder if we should have a method inside Tree that is more descriptive, like `tree.IsEmpty()`

Reply:
Yes, please do. So we can quickly check if we can save safely

Reply:
We only really need to check the Size actually, since if the latest version is > 0, then there are keys in the tree, but I'd still recommend using the new !IsEmpty.